glass/fields.py

```diff
@@ -51,15 +51,25 @@ def wrapper(*args: _P.args, **kwargs: _P.kwargs) -> _R:
         return decorator
 
 
+def _is_inv_triangle_number(triangle_number: int) -> tuple[int, bool]:
+    r"""
+    The :math:`n`-th triangle number is :math:`T_n = n \, (n+1)/2`.  If
+    the argument is :math:`T_n`, then :math:`n` is returned. Otherwise,
+    a :class:`ValueError` is raised.
+    """
+    n = math.floor(math.sqrt(2 * triangle_number))
+    return n, n * (n + 1) // 2 == triangle_number
+
+
 def nfields_from_nspectra(nspectra: int) -> int:
     r"""
     Returns the number of fields for a number of spectra.
 
     Given the number of spectra *nspectra*, returns the number of
     fields *n* such that ``n * (n + 1) // 2 == nspectra``.
     """
-    n = math.floor(math.sqrt(2 * nspectra))
-    if n * (n + 1) // 2 != nspectra:
+    n, is_inv_triangle = _is_inv_triangle_number(nspectra)
```

Comment:
`_is_inv_triangle_number` will similarly be used in a new function (to be added in #533) which will be specific to `alm`.